tfac/dataImport.py

```diff
@@ -165,32 +165,14 @@ def form_limit_tensor():
     sources and cohorts.
 
     Returns:
-        limit_tensor (numpy.ndarray): tensor of detection limits for each
-        cytokine and patient; same shape as the tensor produced via form_tensor
+        limit_tensor (pandas.DataFrame): detection limits for each cytokine
+            across serum/plasma sources; set to be highest detection limit
+            across cohorts
     """
-    plasma_limit, serum_limit = import_cytokines(scale_cyto=False)
-    patient_data = import_patient_metadata()
-    cohorts = set(patient_data.loc[:, 'cohort'])
-
-    patients = set(patient_data.index)
-    serum_limit = add_missing_columns(serum_limit, patients)
-    plasma_limit = add_missing_columns(plasma_limit, patients)
-
-    for cyto in [serum_limit, plasma_limit]:
-        for cohort in cohorts:
-            cohort_patients = patient_data.loc[
-                patient_data.loc[:, 'cohort'] == cohort,
-                :
-            ].index
-            cyto_cohort = cyto.loc[:, cohort_patients]
-            limit = cyto_cohort.min(axis=1)
-
-            for cohort_patient in cohort_patients:
-                cyto.loc[:, cohort_patient] = limit
-
-    limit_tensor = np.stack(
-        (serum_limit.to_numpy(dtype=float), plasma_limit.to_numpy(dtype=float))
-    ).T
+    limit_tensor = np.load(
+        join(PATH_HERE, 'tfac', 'data', 'mrsa', 'LoD_tensor.pkl'),
```

Comment:
I really think this file should be a CSV so that it can easily be reused elsewhere and inspected. This would also allow it to be imported as a pandas table, as the docstring suggests.

Reply:
Sure--I don't think we can store 3d tensors as .csv though. We could move to a multi-indexed dataframe that has LoDs for each cytokine/cohort.